rust-lib/src/lib.rs

```diff
@@ -120,36 +125,64 @@ where
     }
 }
 
-#[derive(Debug, Deserialize)]
-struct StartRgbArgs {
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    network: bp::Chain,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    stash_endpoint: SocketLocator,
-    contract_endpoints: HashMap<ContractName, String>,
-    threaded: bool,
-    datadir: String,
+#[derive(Debug, Display, From, Error)]
+#[display(doc_comments)]
+#[non_exhaustive]
+enum RequestError {
+    /// String error
+    #[from]
+    StringErr(String),
```

Comment:
No need for this generic error type which does not explain the reason behind the error. I will adjust code below in order to avoid having it

Reply:
removing it causes the build to fail:
```
176 |         serde_json::from_str(&ptr_to_string(contract_endpoints)?)?;
    |                                                                ^ the trait `From<std::string::String>` is not implemented for `RequestError`

283 |     let runtime = Runtime::from_opaque(runtime)?;
    |                                                ^ the trait `From<std::string::String>` is not implemented for `RequestError`

285 |     let inputs: Vec<OutPoint> = serde_json::from_str(&ptr_to_string(inputs)?)?;
    |                                                                            ^ the trait `From<std::string::String>` is not implemented for `RequestError`

287 |     let allocate: Vec<Outcoins> = serde_json::from_str(&ptr_to_string(allocate)?)?;
    |                                                                                ^ the trait `From<std::string::String>` is not implemented for `RequestError`
```
I think we need to change the `ptr_to_string` function (and consequently the `_issue` function), and the `from_opaque`  function. I'll try and then let you know

Reply:
done, I hope it's ok. I requested a review on a point I'm not sure about